ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -697,7 +700,49 @@ def yield_view_lineage(self) -> Iterable[Either[OMetaLineageRequest]]:
         else:
             yield from self._process_view_def_serial()
 
-    def _get_foreign_constraints(self, foreign_columns) -> List[TableConstraint]:
+    def _is_column_unique(self, column: Dict, columns: List[Column]) -> bool:
```

Comment:
why is `Column` only a dict? In any case, this method should have unit tests. Might even be a good candidate for leaving the class since I don't see `self` being used. easier to test that way as well

Reply:
The `Column` here comes from sqlchemy's column response. Ref [here](https://github.com/open-metadata/OpenMetadata/pull/18021/files#diff-59f7cdd6f67326202a9b9326a4e693801f8369376ca3b7144756a24eff4b9cc6R783)

I'll add these methods in separate module